client/state/initial-state.js

```diff
@@ -87,16 +88,29 @@ export function persistOnChange( reduxStore, serializeState = serialize ) {
 	return reduxStore;
 }
 
+// This can't be a middleware as those are invoked before the reducer that
```

Comment:
>I haven't, but I sure like the idea :-)

Or perhaps even something that could be built in to `<DocumentHead />`, with one instance rendered toward top of the hierarchy tree without title attributes, only to keep title in sync with state? Not sure that's reasonable vs. separate components, but worth considering.